website/src/controllers/api-v3/user.js

```diff
@@ -31,4 +38,45 @@ api.getUser = {
   },
 };
 
+/**
+ * @api {post} /email/update
+ * @apiVersion 3.0.0
+ * @apiName EmailUpdate
+ * @apiGroup User
+ *
+ * @apiSuccess {Object} { status: 'ok' }
+ **/
+api.updateEmail = {
+  method: 'POST',
+  middlewares: [authWithHeaders(), cron],
+  url: '/email/update',
+  async handler (req, res) {    
+    let user = res.locals.user.toJSON();
+    
+    console.log("+++ +++ user._id is", user._id);
+    user = await User.findOne({ _id: user._id }).exec();
+    console.log('+++ +++ user.auth.local is', user.auth.local);
+
+    if (!user.auth.local.email) throw new PreconditionFailed(res.t('userHasNoLocalRegistration'));
+
+    req.checkBody('newEmail', res.t('newEmailRequired')).notEmpty().isEmail();
+    req.checkBody('password', res.t('missingPassword')).notEmpty();
+    let validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+ 
+    // check password
+    let temp_user = await User.findOne({ _id: user._id}, {"auth.local": 1}).exec();
+    let candidate_password = passwordUtils.encrypt(req.body.password, temp_user.auth.local.salt);
+    if (candidate_password !== temp_user.auth.local.hashed_password) throw new BadRequest();
```

Comment:
Yes... I specifically would like to make the authentication errors not precise. If password doesn't match, I'd like the error to be the same as username not found (because otherwise you can run an attack to find all the usernames). That's why authentication errors can be made less clear.